Userland/Libraries/LibJS/Heap/BlockAllocator.cpp

```diff
@@ -16,7 +16,7 @@
 #endif
 
 // FIXME: Implement MADV_FREE and/or MADV_DONTNEED on SerenityOS.
-#if defined(AK_OS_SERENITY) || (!defined(MADV_FREE) && !defined(MADV_DONTNEED))
+#if defined(AK_OS_SERENITY) || defined(AK_OS_GNU_HURD) || (!defined(MADV_FREE) && !defined(MADV_DONTNEED))
```

Comment:
Any reason these don't exist on the Hurd other than "not yet implemented"? Would be unfortunate to keep around the old code path only for Hurd, it would likely get stale.

Reply:
If you're expecting this to drop/zero the page lazily, when the VM subsystem is out of memory, then I don't think Mach has equivalent functionality, so there isn't much the Hurd could do here (other than perhaps us implementing this into GNU Mach 🤔).

If you want to zero out the page immediately, as Linux does, then why don't you just munmap+map? `MADV_DONTNEED` is a really weird interface for that.

Reply:
The purpose of the new API is to keep the virtual memory reserved forever. Very 32-bit unfriendly :)

Reply:
But since you're mapping with `MAP_FIXED`, you'll get back exactly the same virtual memory addresses — in fact you don't even need to `munmap()`, and it'll be atomic even (but not on the Hurd, sadly).

Also I don't see how it's 32-bit unfriendly, unless of course you're planning on WebContent taking up gigabytes of memory 😉 ASLR-unfriendly, perhaps.